modules/tinymce/src/plugins/lists/test/ts/browser/ListPropertiesTest.ts

```diff
@@ -1,137 +1,136 @@
+import { SugarBody, SugarDocument, SugarElement, Value } from '@ephox/sugar';
+import { assert } from 'chai';
 
 import Editor from 'tinymce/core/api/Editor';
 import Plugin from 'tinymce/plugins/lists/Plugin';
 import Theme from 'tinymce/themes/silver/Theme';
 
-UnitTest.asynctest('browser.tinymce.plugins.lists.ListPropertiesTest', (success, failure) => {
-  Plugin();
-  Theme();
+describe('browser.tinymce.plugins.lists.ListPropertiesTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'lists',
+    contextmenu: 'lists bold',
+    menu: {
+      custom: { title: 'Custom', items: 'listprops' }
+    },
+    menubar: 'custom',
+    toolbar: false,
+    indent: false,
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ Plugin, Theme ], true);
 
   const contentMenuSelector = '.tox-tinymce-aux .tox-menu .tox-collection__item:contains("List properties...")';
   const inputSelector = 'label:contains(Start list at number) + input.tox-textfield';
 
-  const sOpenContextMenu = (editor: Editor, tinyUi: TinyUi, target: string) => Chain.asStep(editor, [
-    tinyUi.cTriggerContextMenu('trigger context menu', target, '.tox-silver-sink [role="menuitem"]'),
-    Chain.wait(0)
-  ]);
+  const openContextMenu = async (editor: Editor, selector: string) => {
+    Mouse.contextMenuOn(TinyDom.body(editor), selector);
+    await TinyUiActions.pWaitForUi(editor, '.tox-silver-sink [role="menuitem"]');
+  };
 
-  const sOpenDialog = (editor: Editor, tinyUi: TinyUi, contextMenuSelector: string) => {
-    const doc = SugarElement.fromDom(document);
-    return GeneralSteps.sequence([
-      sOpenContextMenu(editor, tinyUi, contextMenuSelector),
-      FocusTools.sTryOnSelector('Wait for the context menu to be focused', doc, contentMenuSelector),
-      Keyboard.sKeydown(doc, Keys.enter(), {}),
-      tinyUi.sWaitForUi('Wait for the dialog to appear', '[role=dialog]')
-    ]);
+  const openDialog = async (editor: Editor, contextMenuSelector: string) => {
+    await openContextMenu(editor, contextMenuSelector);
+    Keyboard.activeKeydown(SugarDocument.getDocument(), Keys.enter(), {});
+    await TinyUiActions.pWaitForUi(editor, '[role=dialog]');
   };
 
-  const sUpdateStart = (currentValue: string, newValue: string) => {
-    const doc = SugarElement.fromDom(document);
-    return GeneralSteps.sequence([
-      FocusTools.sIsOnSelector('Check focus is on the input field', doc, inputSelector),
-      Chain.asStep(doc, [
-        FocusTools.cGetFocused,
-        Chain.op((input) => {
-          Assertions.assertEq('Assert initial start value', currentValue, Value.get(input));
-        }),
-        UiControls.cSetValue(newValue)
-      ])
-    ]);
+  const updateDialog = (currentValue: string, newValue: string) => {
+    const doc = SugarDocument.getDocument();
+    FocusTools.isOnSelector('Check focus is on the input field', doc, inputSelector);
+    const input = FocusTools.getFocused(doc).getOrDie();
```

Comment:
```suggestion
    const input = FocusTools.getFocused<HTMLInputElement>(doc).getOrDie();
```

Reply:
As discussed this morning, we should improve `getFocused` to allow us to assert the type more easily.